plugins/utils/duration.go

```diff
@@ -0,0 +1,53 @@
+/****************************************************************************
+ * Copyright 2023, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package utils //
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// Duration represents json unmarshallable implementation of time.Duration
+type Duration struct {
+	time.Duration
+}
+
+// UnmarshalJSON unmarshal json bytes to time duration
+func (duration *Duration) UnmarshalJSON(b []byte) error {
+	var unmarshalledJSON interface{}
+
+	err := json.Unmarshal(b, &unmarshalledJSON)
+	if err != nil {
+		return err
+	}
+
+	switch value := unmarshalledJSON.(type) {
+	case float64:
+		duration.Duration = time.Duration(value)
```

Comment:
Do we accept number without unit? What is the default unit for this case?

Reply:
yes we do, default unit is nanosecond by golang.